app/Http/Controllers/APIDungeonRouteController.php

```diff
@@ -167,6 +167,10 @@ function rate(Request $request, DungeonRoute $dungeonroute)
     {
         $value = $request->get('rating', -1);
 
+        if ($dungeonroute->isOwnedByCurrentUser()) {
+            return response()->setStatusCode(403);
```

Comment:
This is more of a personal thing, but I prefer to have a single point of exit in my code which is located at the end of the function. Unless it'd really make my code ugly I try to adhere to it. I think it's best if you try the same, unless you got a good argument against it of course. Otherwise I'm just gonna change it myself since I've been doing it for so long now :)

Reply:
I usually place most of my constraints at the beginning to prevent any unnecessary database calls etc. So if anything I'd personally move it up to go above the `$value` assignment.

Reply:
What I try to do (but not always I guess) is have a clear happy flow, then put any exceptions in elses. Example:

```
if( happy1 ){
    if( happy2) {
        happy();
    } else {
        error2();
    }
} else {
    error1();
}
```
In the end you return the result below all that. I dislike having to find a bunch of return paths once functions start to grow bigger where it turns out that halfway through are a bunch of return statements.

Of course as you say you should exit prior to doing unnecessary database updates. I just prefer to if chain with a single point of return rather than a return at the beginning :).

I'm not bashing your code by the way, I'm trying to find a common ground for both of us so we continue in the same style etc, otherwise we're going to end up diverging and that's never good.

Reply:
Btw, I'm going to have to push my local changes now as I'm about to leave work, I'm quite sure you're going to get merge conflicts since I already fixed this bug. As I said before, I'll start working more organized so that this won't happen too often in the future.

Reply:
I did some more thinking, and I think it's okay to put your constraints at the top of the function and return then. Just because I wouldn't do it that way shouldn't mean you can't. It's not a bad practice by any standard so it's fine :)